### DISCUSSION TITLE
filter for allowed dataobjects in many-to-many object relation

### DISCUSSION THREAD

**Discussion by langert** [pimcore/pimcore#16852] [Q&A / Support]
Hi,

is there a way to set a filter for dataobject to be able to link in an many-to-many object relation? I can define the classes for objects but no filter within the objects of the selected classes.

E.g. only objects from the defines classes should be allowed, where attributXy = true.

I could instead use a select option and build a customer OptionsProvider but a many-to-many object relation seems more strait forward to me and provides the drag and drop possibility for users...

Any ideas on that?

**Comment by AlephLau** [comment]
@langert  As you mentioned, it's indeed possible to restrict the linking of a relationship based on the object type, but there isn't an integrated way in Pimcore to additionally discriminate based on an attribute in the object. The alternative you suggest seems very good to me, using an option provider in a multi-select, as this way the user knows what they can select. In the other way, with drag and drop, the user wouldn't know what type of object they can relate to until they check whether the attribute you want to filter by is allowed or not.

Certainly, when searching for permitted objects to add in a relationship, you can indeed utilize filters that perform the same function as you would with an option provider. These filters can help narrow down the selection based on specific criteria, including attributes within the objects. This can provide a more streamlined and user-friendly experience compared to relying solely on drag and drop functionality.
![image](https://github.com/pimcore/pimcore/assets/70108013/197729b7-8dad-4e45-b918-a15a7715def6)

**Reply by langert** [reply]
@AlephCode I like the idea using filters for searching objects and perhaps combine it with an eventlistener to throw an exception before saving objects when an object is choosen that I do not want to be allowed as a relation.
